Adding B2B release notes (#7566)
magento/devdocs: src/guides/v2.4/release-notes/b2b-release-notes.md

group: release-notes
title: B2B Release Notes
---

The release notes for the B2B extension captures additions and bug fixes that Magento has added during a release cycle.
These release notes can include:

-  {:.new}New features
-  {:.fix}Fixes and improvements

> ## Magento B2B - Version 2.4.0

Author: Good catch. Fixed.

> -  {:.new} Purchase Orders have been enhanced and rewritten. The are now included my default in Magento Commerce.

Reviewer: ```suggestion
-  {:.new} Purchase Orders have been enhanced and rewritten. The are now included by default in Magento Commerce.
```

Use a variable instead of Magento Commerce. Make sure the Jeanne knows that the feature has been transferred to Commerce.

> -  {:.new} Purchase Orders have been enhanced and rewritten. The are now included by default in Magento Commerce.

Reviewer 2: The should be 'They' in 2nd sentence

Author: Good catch

> -  {:.new} Purchase Order Approval Rules have been implemented. This allows users to control the Purchase Order workflow by creating purchasing rules for orders.

Reviewer: ```suggestion
-  {:.new} Purchase Order Approval Rules have been implemented. These rules allow users to control the Purchase Order workflow by creating purchasing rules for orders.
```

-  {:.new} Login as Customer is now included by default in Magento Commerce. This allows site employees to assist customers by logging in as the customer to see what they see.

-  {:.fix} Attribute aggregations are now working correctly for Layered Navigation with Elasticsearch

> -  {:.fix} Search orders by special chars is now working properly.

Reviewer 2: write out chars as characters

-  :.fix} Clicking on &quot;Clear All&quot; Button now expands all the filters, rather than collapsing them.
-  {:.fix} Product SKU/Name is now included in the Order History search filter summary.
-  {:.fix} Sorting Indicator is now displayed properly in the My Purchase Orders Grid.

> -  {:.fix} 'https://jira.corp.magento.com/browse/B2B-564'>B2B-564 - Now, you can click the Approve/Cancel/Reject/Purchase Purchase Order button only once. Previously, you could click the button multiple times.

Reviewer: Delete the URL and issue number, I think.

> -  {:.fix} The Purchase Order Approve, Reject, Cancel, Validate buttons now render correctly on mobile devices.

Reviewer: In the previous item, the buttons were separated by slashes. If you change the separator to commas, add "and" before the last button,

> -  {:.fix} Previously, approving a Purchase Order with a discount that has expired placed the order at the full amount &amp; not update the Purchase Order total. This now works correctly.

Reviewer: ```suggestion
-  {:.fix} Previously, approving a Purchase Order with a discount that has expired placed the order at the full amount and did not update the Purchase Order total. This now works correctly.
```

> -  {:.fix} A bug was introduced in 2.3.4 where an custom extension attributes would not be copied from the Customer Address to the Quote Address. This bug has been fixed.

Reviewer: ```suggestion
-  {:.fix} A bug was introduced in 2.3.4 where custom extension attributes would not be copied from the Customer Address to the Quote Address. This bug has been fixed.
```

> -  {:.fix} With B2B installed, a SQL error would appear when assigning categories to shared catalogs. This bug has been fixed.

Reviewer: ```suggestion
-  {:.fix} With B2B installed, an SQL error would appear when assigning categories to shared catalogs. This bug has been fixed.
```

Author: Keeping "a".

-  {:.fix} Due to an incorrect variable type value, admins could not add configurable products to an order. The option dropdowns would not populate. This feature now works properly.
-  {:.fix} Previously, when editing Category Permissions for the Not Logged In group, an error would occur when saving the changes. This bug has now been fixed.
-  {:.fix} A bug has been fixed that now allows admins to add products to an order that are not in the shared catalog. Previously, an error message would appear when adding an item not in the catalog.

> -  {:.fix} Previously, after running the command `php bin/magento indexer:set-dimensions-mode catalog_product_price website` and then trying to make a shared catalog, an error would occur. This bug has been fixed.

Reviewer 2: suggest changing 'make' to 'create' because this is how we normally describe the action

> -  {:.fix} When adding a new company and assigning it a non-default website, the wrong site id was sent, causing an error. This bug has now been fixed.

Reviewer 2: This one as written is very confusing. Companies are global entities, so when creating a new company the admin user does not assign it to any website. However, users within a company are assigned to a website. When creating a new company the admin user is prompted to fill in details for the company administrator. One of the fields in this section is 'Website'. 

The original bug was if you created a new company and assigned the _company administrator_ to any website that was not the default website, 'the wrong ID was sent, causing an error.'

Propose changing to:
`When adding a new company and assigning the Company Administrator to a non-default website, the wrong site ID was sent, causing an error. This bug has now been fixed.`

-  {:.fix} Previously, after a customer was moved to another customer group, adding a product to an order via Quick View would fail with an error. This bug has now been fixed.

> -  {:.fix} Previously, when attempting to checkout use the WebAPI with a B2B quote, an incorrect value was sent to the API, causing an error to occur. This bug has been fixed.

Reviewer: ```suggestion
-  {:.fix} Previously, when attempting to checkout using the WebAPI with a B2B quote, an incorrect value was sent to the API, causing an error to occur. This bug has been fixed.
```

-  {:.fix} Previously, when setting a company to "Active" via the API, an error would occur. This bug has now been fixed.
-  {:.fix} Due to an unneeded `form` tag, the page would refresh when you press Enter after changing a proposed shipping charge. This has been fixed.
-  {:.fix} Previously, when setting a product display limit on a catalog page and that limit was less than the number of total products, an error occurred. That feature now works as expected.

> -  {:.fix} Previously, when changing the admin of a company, the original admin address would be copied to the new admin, giving them 2 addresses. Now, only the correct address is added.

Reviewer: ```suggestion
-  {:.fix} Previously, when changing the admin of a company, the original admin address would be copied to the new admin, giving them two addresses. Now, only the correct address is added.
```
